frontend/src/components/Admin/CoffeeChats/CoffeeChatDashboard.tsx

```diff
@@ -13,23 +14,51 @@ type CoffeeChatStats = {
 };
 
 const CoffeeChatDashboard: React.FC = () => {
-  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
   const [bingoBoard, setBingoBoard] = useState<string[][]>([]);
   const [coffeeChats, setCoffeeChats] = useState<CoffeeChat[]>([]);
+  const [isNewbieMap, setIsNewbieMap] = useState<{ [key: string]: boolean }>({});
 
   const allMembers = useMembers();
 
   useEffect(() => {
-    CoffeeChatAPI.getCoffeeChatBingoBoard().then((board) => {
-      setBingoBoard(board);
-    });
+    const fetchData = async () => {
+      setIsLoading(true);
 
-    CoffeeChatAPI.getAllCoffeeChats().then((chats) => {
-      setCoffeeChats(chats.filter((chat) => chat.status === 'approved'));
-    });
+      const boardPromise = CoffeeChatAPI.getCoffeeChatBingoBoard().then((board) => {
+        setBingoBoard(board);
+      });
+
+      const chatsPromise = CoffeeChatAPI.getAllCoffeeChats().then((chats) => {
+        setCoffeeChats(chats.filter((chat) => chat.status === 'approved'));
+      });
+
+      const fetchNewbieStatus = async () => {
+        const newbieMap: { [key: string]: boolean } = {};
+
+        await Promise.all(
+          allMembers.map(async (member) => {
+            const { status } = await CoffeeChatAPI.checkMemberMeetsCategory(
+              member,
+              member,
+              'a newbie'
```

Comment:
instead of this check, can we use `member.semesterJoined`? One thing we should be very careful about is that we shouldn't use the bingo board categories anywhere except for the validation logic `checkMemberMeetsCategory`. We shouldn't even hardcode it anywhere except for the validation logic because the categories are going to change next semester, and there's no guarantee that we'll continue to store any of the data.